src/components/todo-list-element-builder.components.ts

```diff
@@ -0,0 +1,270 @@
+    const events: EventsPayload[] = [
+      {
+        type: 'compositionstart',
+        handler: () => (isComposing = true),
+      },
+      {
+        type: 'compositionend',
+        handler: () => (isComposing = false),
+      },
+      {
+        type: 'input',
+        handler: (e: Event) => {
+          const value = (e.target as HTMLInputElement).value;
+          this.dispatch({
+            type: EVENT_BUS_TYPES.CHANGE_INPUT_VALUE,
+            payload: value,
+          });
+        },
+      },
+      {
+        type: 'keydown',
+        handler: (e: Event) => {
+          if ((e as KeyboardEvent).key === 'Enter') {
+            if (isComposing) {
+              return;
+            }
+            this.dispatch({
+              type: EVENT_BUS_TYPES.ADD_ITEM,
+            });
+            (e.target as HTMLInputElement).value = '';
+          }
+        },
+      },
+    ];
+    const input = this.createInput(events);
+    input.className = this.todoStyles.clsNames.input;
+    input.placeholder = this.options.placeholder || 'What needs to be done?';
+    return input;
+  }
+
+  onChangeCheckbox(e: Event) {
+    this.dispatch({
+      type: EVENT_BUS_TYPES.CHANGE_CHECK,
+      payload: {
+        id: this.getLiDataId(e),
+      },
+    });
+  }
+
+  createTodoListElement() {
+    const events: EventsPayload[] = [
+      {
+        type: 'change',
+        handler: (e: Event) => this.onChangeCheckbox(e),
+      },
+    ];
+    const ul = this.createUl(events);
+    ul.className = `${this.todoStyles.clsNames.ul}`;
+    return ul;
+  }
+
+  createTodoToolboxElement() {
+    const wrapper = document.createElement('div');
+    const buttonPack = document.createElement('div');
+    const { allItems, activeItems, completedItems } = this.initialLabel;
+
+    wrapper.className = this.todoStyles.clsNames.buttonWrapper;
+    buttonPack.className = this.todoStyles.clsNames.buttonPack;
+
+    // item count
+    const elItemCnt = this.createItemCntLabel(this.initialLabel.itemCnt, 0);
+
+    // all items
+    const elAllItems = this.createAllItemsButton(allItems);
+
+    // active items
+    const elActiveItems = this.createActiveItemsButton(activeItems);
+
+    // completed items
+    const elCompletedItems = this.createCompletedItemsButton(completedItems);
+
+    // clear completed
+    const elClearCompleted = document.createElement('div');
+    elClearCompleted.className = this.todoStyles.clsNames.clear;
+    elClearCompleted.appendChild(
+      this.createClearCompletedButton(this.initialLabel.clearCompleted, 0),
+    );
+
+    return {
+      wrapper,
+      buttonPack,
+      elItemCnt,
+      elAllItems,
+      elActiveItems,
+      elCompletedItems,
+      elClearCompleted,
+    };
+  }
+
+  getLiDataId(e: Event) {
+    const dataId: string | null =
+      (e.target as HTMLInputElement | HTMLLabelElement).closest('li')?.getAttribute('data-id') ||
+      null;
+    if (!dataId) {
+      return;
+    }
+    return dataId;
```

Comment:
```suggestion
  getLiDataId({ target }: Event) {
    return (target as HTMLInputElement | HTMLLabelElement).closest('li')?.getAttribute('data-id') || undefined;
```
타입이 자동추론되니 이렇게 줄일 수 있을듯합니다.